### PR TITLE
Searchcollection

### PR DESCRIPTION
**Pull request by yuki617**

### CODE REVIEW

File: pyserini/search/pysearch.py [hunk 1]
```diff
@@ -18,6 +18,10 @@
 This module provides Pyserini's Python search interface to Anserini. The main entry point is the ``SimpleSearcher``
 class, which wraps the Java class with the same name in Anserini.
 """
+import re
+import argparse
+import sys
+sys.path.insert(0,'./')
```
Comment (lintool) [inline pyserini/search/pysearch.py:24]: is this necessary? I don't think so...

File: pyserini/search/pysearch.py [hunk 2]
```diff
@@ -368,4 +372,35 @@ def multisearch(self, q: str, k=10) -> List[List[JSimpleNearestNeighborSearcherR
         List(List[JSimpleNearestNeighborSearcherResult])
             List of List of (nearest neighbor) search results (one for each matching id).
         """
-        return self.object.multisearch(JString(q), k)
\ No newline at end of file
+        return self.object.multisearch(JString(q), k)
+
+
+def main(index, topics, output):
```
Comment (lintool) [inline pyserini/search/pysearch.py:378]: Move this down into `__main__`

File: pyserini/search/pysearch.py [hunk 3]
```diff
@@ -368,4 +372,35 @@ def multisearch(self, q: str, k=10) -> List[List[JSimpleNearestNeighborSearcherR
...
+        hits = searcher.search(search, 1000)
+        for i in range(0, len(hits)):
+            my_list.append(f'{number} Q0 {hits[i].docid.strip()} {i + 1} {hits[i].score:.6f} Anserini')
+    with open(output, 'w') as f:
+        for item in my_list:
+            f.write("%s\n" % item)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Create a ArcHydro schema')
```
Comment (lintool) [inline pyserini/search/pysearch.py:398]: "ArcHydro schema"??

File: pyserini/search/pysearch.py [hunk 4]
```diff
@@ -368,4 +372,35 @@ def multisearch(self, q: str, k=10) -> List[List[JSimpleNearestNeighborSearcherR
...
\ No newline at end of file
+        return self.object.multisearch(JString(q), k)
+
+
+def main(index, topics, output):
+    print(index, topics, output)
+    searcher = SimpleSearcher(index)
+    with open(topics, 'r') as content_file:
+        content = content_file.read()
+    result = re.findall(r'(?<=<top>)(.+?)(?=<desc>)', content, flags=re.S)
```
Comment (lintool) [inline pyserini/search/pysearch.py:383]: You can load topics directly: https://github.com/castorini/pyserini/blob/master/tests/test_loadtopics.py

So allow something like `--topics robust04`?

Comment (yuki617) [reply pyserini/search/pysearch.py:383]: should I keep allowing the -topics filepath one?

Comment (lintool) [reply pyserini/search/pysearch.py:383]: interpret it as an id, per above, otherwise interpret it as a path.

Comment (yuki617) [reply pyserini/search/pysearch.py:383]: got it. thanks

File: pyserini/search/pysearch.py [hunk 5]
```diff
@@ -375,27 +373,36 @@ def multisearch(self, q: str, k=10) -> List[List[JSimpleNearestNeighborSearcherR
...
-    with open(output, 'w') as f:
-        for item in my_list:
-            f.write("%s\n" % item)
-
 
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser(description='Create a ArcHydro schema')
+
+    def main(index, topics, output):
+        searcher = SimpleSearcher(index)
```
Comment (lintool) [inline pyserini/search/pysearch.py:380]: I don't think we need an extra method definition? just move the code down?

File: pyserini/search/pysearch.py [hunk 6]
```diff
@@ -375,27 +373,36 @@ def multisearch(self, q: str, k=10) -> List[List[JSimpleNearestNeighborSearcherR
...
+        my_list = []
+        if topics == 'robust04':
+            topics_dic = get_topics('robust04')
```
Comment (lintool) [inline pyserini/search/pysearch.py:383]: I think you can just do something like `get_topics(topics)` and check `None`?

File: pyserini/search/pysearch.py [hunk 7]
```diff
@@ -375,27 +373,36 @@ def multisearch(self, q: str, k=10) -> List[List[JSimpleNearestNeighborSearcherR
...
+            topics_lst = re.findall(r'(?<=<top>)(.+?)(?=<desc>)', content, flags=re.S)
+            topics_lst = [i.strip() for i in topics_lst]
+            for topic in topics_lst:
+                number_search = topic.split("<title>")
+                number = number_search[0].strip().split(" ")[-1]
+                search = number_search[1].strip()
+                hits = searcher.search(search, 1000)
+                for i in range(0, len(hits)):
+                    my_list.append(f'{number} Q0 {hits[i].docid.strip()} {i + 1} {hits[i].score:.6f} Anserini')
+        with open(output, 'w') as f:
```
Comment (lintool) [inline pyserini/search/pysearch.py:401]: I think you can move the `with` to higher scope, you don't need to append to `my_list`, and just directly write out?

**Pull request merged**